flyflix.py

```diff
@@ -63,11 +65,41 @@ def read_metadata():
         try:
             with metadata_lock:
                 metadata = yaml.safe_load(stream)
```

Comment:
Since you know that the result from `yaml.safe_load()` is potentially not what you want and might contain "unsafe" content, it would be good not to store the content in the global variable.

Why don't you rewrite the `metadata_as_string()` function to accept an input dictionary and return a resulting dictionary? In that case you would encapsulate the modification and that function would not have to be specific for the `metadata`.

In general, using global variables is bad practice and should be avoided as much as possible (yes, I know I did that as well…:-)) If you are using global variables, try to minimize that to as few places as possible. Otherwise you might loose track of what the global variable contains at any point in time.


e are going to modify the `metadata` variable later,